packages/labs/ssr/src/lib/render-result-readable.ts

```diff
@@ -56,14 +57,16 @@ export class RenderResultReadable extends Readable {
     // result, because we must be able to return in the middle of the loop
     // and resume on the next call to _read().
 
-    // Get the current iterator
-    let iterator = this._iterators.pop();
+    // Get the current iterator, only if we don't already have one from the previous call to _read()
+    if (!this._currentIterator) {
+      this._currentIterator = this._iterators.pop();
+    }
```

Comment:
nits: wrap comments at 80 cols and use nullish-coalescing assignment:

```suggestion
    // Get the current iterator, only if we don't already have one from the
    // previous call to _read()
    this._currentIterator ??= this._iterators.pop();
```